File: Archive/mad/prg/Exercises/Ej_5.5/src/Arrays.cpp
#include <iostream>

using namespace std;

int* generateInputArray(int* length){
	cout << "Inserte tamaño de la array: ";
	cin >> *length;
	
	cout << "Inserte el valor mínimo: ";
	int mn;
	cin >> mn;
	cout << "Inserte el valor máximo: ";
	int mx;
	cin >> mx;
	if (mn > mx){
		int tmp = mx;
		mx = mn;
		mn = tmp;
	}
	int* array = new int[*length];
	for (int i = 0; i < *length; i++){
		array[i] = rand() % (mx - mn + 1) + mn;
	}
	
	return array;
}

int* getInputArray(int* length){
	cout << "Inserte tamaño de la array: ";
	cin >> *length;
	
	cout << "Inserte array: ";
	int* array = new int[*length];
	for (int i = 0; i < *length; i++){
		cin >> array[i];
	}
	
	return array;
}

int* removeNumber(int number, int* inArray, int* length){
	int count = 0;
	for (int i = 0; i < *length; i++){
		if (inArray[i] == number) count++;
	}
	
	int newLength = *length - count;
	int* newArray = new int[newLength];
	int c = 0;
	for (int i = 0; i < *length; i++){
		if (inArray[i] != number){
			newArray[c] = inArray[i];
			c++;
		}
	}
	
	delete[] inArray;
	
	*length = newLength;
	return newArray;
}

void printArray(int* array, int length){
	cout << "Nueva array: ";
	for (int i = 0; i < length; i++){
		cout << array[i] << " ";
	}
}

bool continueRemoving(){
	cout << endl << "¿Continuar? (y/n): ";
	string line;
	do{
		getline(cin, line);
	}
	while(line.empty());
	char ch = line.at(0);
	return (ch == 'y' || ch == 's');
}

int main(){
	int length;
	int* array = generateInputArray(&length);
	
	printArray(array, length);
	
	do{
		int number;
		cout << endl << "Indique el número a borrar: ";
		cin >> number;
		
		array = removeNumber(number, array, &length);
		
		printArray(array, length);
	}
	while(length > 0 && continueRemoving());
	
	delete[] array;
	return 0;
}

File: Archive/mad/prg/Exercises/Ej_5.5/src/Arrays.java
import java.util.Scanner;
import java.lang.Math;

public class Arrays{
	
	static int[] generateInputArray(Scanner in){
		System.out.print("Inserte tamaño de la array: ");
		int length = in.nextInt();
		int[] inArray = new int[length];
		
		System.out.print("Inserte el valor mínimo: ");
		int min = in.nextInt();
		System.out.print("Inserte el valor máximo: ");
		int max = in.nextInt();
		if (min > max){
			int tmp = max;
			max = min;
			min = tmp;
		}
		for (int i = 0; i < length; i++){
			inArray[i] = (int) Math.round(Math.random() * (max - min + 1) + min);
		}
		
		return inArray;
	}
	
	static int[] getInputArray(Scanner in){
		System.out.print("Inserte tamaño de la array: ");
		int length = in.nextInt();
		int[] inArray = new int[length];
		
		System.out.print("Inserte array: ");
		for (int i = 0; i < length; i++){
			inArray[i] = in.nextInt();
		}
		
		return inArray;
	}
	
	static int[] removeNumber(int[] inArray, int number){
		int count = 0;
		for (int val : inArray){
			if (val == number)
				count++;
		}
		
		int[] newArray = new int[(inArray.length - count)];
		int c = 0;
		for (int val : inArray){
			if (val != number){
				newArray[c] = val;
				c++;
			}
		}
		
		return newArray;
	}
	
	static void printArray(int[] array){
		System.out.print("Nueva array: ");
		for (int val : array){
			System.out.print(val);
			System.out.print(" ");
		}
	}
	
	static boolean continueRemoving(Scanner in){
		System.out.print("\n¿Continuar? (y/n): ");
		String line;
		do{
			line = in.nextLine();
		}
		while (line.isEmpty());
		char ch = line.toLowerCase().charAt(0);
		return (ch == 'y' || ch == 's');
	}
	
	public static void main(String[] args){
		Scanner in = new Scanner (System.in);
		
		int[] array = generateInputArray(in);
		
		printArray(array);
		
		do{
			System.out.print("\nIndique el número a borrar: ");
			int number = in.nextInt();
			
			array = removeNumber(array, number);
			
			printArray(array);
		}
		while(array.length > 0 && continueRemoving(in));
	}
}